### ISSUE TITLE
Boolean options without a value (--http2) are silently ignored

### ISSUE THREAD

**Issue by mac-chaffee** [joseluisq/static-web-server#209]
**Describe the bug**

Boolean options like `--http2` and `--security-headers` only work if you add `=true` to the end, which is kinda non-standard. Seems the blame lies with [clap's peculiar decision](https://github.com/clap-rs/clap/issues/1649) to ignore decades of option-parsing precedence by default. But also it seems that clap v2.0 (which comes from structopt) does not require the user to supply the `=true` and instead will just silently ignore the flag.

**To Reproduce**

```
$ static-web-server --port 8000 --log-level=trace --security-headers --root=./ 2>&1 | grep "security headers"
2023-05-29T16:19:43.950269Z  INFO static_web_server::server: security headers: enabled=false

$ static-web-server --port 8000 --log-level=trace --security-headers=true --root=./ 2>&1 | grep "security headers"
2023-05-29T16:21:45.744096Z  INFO static_web_server::server: security headers: enabled=true

$ static-web-server --port 8000 --log-level=trace --http2 --root=./ 2>&1 | grep "security headers"
2023-05-29T16:21:04.107193Z  INFO static_web_server::server: security headers: enabled=false
(also should have crashed since no tls cert flags were provided)
```

**Expected behavior**
* `--http2` should imply "true", or `--http2` should throw an error that a value is required.

**Environment and Specs**
 - static-web-server: 2.16.0 (built from commit 0f66443)
 - OS: Linux

**Additional context**
If we migrate from structopt to clapv3 (which includes structopt), then it will log `error: The argument '--http2 <HTTP2>' requires a value but none was supplied`. But I'm struggling to implement the "--http2 implies true, no value required" behavior.

I have started down this path of migrating to clapv3 here: https://github.com/static-web-server/static-web-server/compare/master...mac-chaffee:static-web-server:clap-migration

**Comment by joseluisq** [comment]
Yeah, this is a dependency issue regarding ignoring boolean options behavior without an explicit value.
But I can say that this is also something not supported on `structopt` as far as I remember in https://github.com/clap-rs/clap/issues/1649 (as you referred).

Anyway, I realized that at this point, [`structopt` is sort of deprecated](https://github.com/TeXitoi/structopt/issues/525) since it is included in `clap` v3+. So a migration path should be followed.

**Comment by joseluisq** [comment]
> If we migrate from structopt to clapv3 (which includes structopt), then it will log `error: The argument '--http2 <HTTP2>' requires a value but none was supplied`. But I'm struggling to implement the "--http2 implies true, no value required" behavior.
> 
> I have started down this path of migrating to clapv3 here: [master...mac-chaffee:static-web-server:clap-migration](https://github.com/static-web-server/static-web-server/compare/master...mac-chaffee:static-web-server:clap-migration)

Do you have a plan to submit it as a PR or draft? I think should be great, then I could try to help.

**Comment by joseluisq** [comment]
Released on [v2.17.0](https://github.com/static-web-server/static-web-server/releases/tag/v2.17.0)

**Comment by joseluisq** [comment]
Improved on [v2.18.0](https://github.com/static-web-server/static-web-server/releases/tag/v2.18.0)
